zone/combat_record.h

```diff
@@ -0,0 +1,22 @@
+#ifndef EQEMU_COMBAT_RECORD_H
+#define EQEMU_COMBAT_RECORD_H
+
+#include <ctime>
+#include <string>
+
+class CombatRecord {
+public:
+	void Start(std::string in_mob_name);
+	void Stop();
+	bool InCombat();
+	void ProcessHPEvent(int hp, int current_hp);
+	double TimeInCombat() const;
+private:
+	std::string mob_name;
+	time_t      start_time      = 0;
+	time_t      end_time        = 0;
+	__int64_t   damage_received = 0;
```

Comment:
we have int64

Reply:
That's fine but its the same thing under the hood

```
typedef __int64_t int64_t;
```

```
typedef int64_t int64;
```

It's just aliases references

Reply:
Consistency with the rest of the code base is good though :P

Reply:
The rest of the code uses this same type under the hood. Pointless to pull in a header file `types.h` for what is essentially the same thing for one field on one struct

Reply:
https://en.cppreference.com/w/cpp/header/cstdint Is what I would actually prefer we use, but that's a lot of changes and never really brought it up

Reply:
That's fine